scenes/players.gd

```diff
@@ -0,0 +1,26 @@
+extends Node
+class_name Players
+const player_factory = preload("res://scenes/characters/player.tscn")
+signal player_added_to_scene(peer_id, player)
+@export var starting_marker: Marker2D
+
+func add_player_to_scene(peer_id: int) -> void:
+	var player: Player = player_factory.instantiate()
+	player.set_name(str(peer_id))
+	add_child(player)
+	if starting_marker != null:
+		player.position = starting_marker.position
+	else:
+		player.position = Vector2(0, 0)
+	player_added_to_scene.emit(peer_id, player)
+
+func remove_player_from_scene(peer_id: int) -> void:
+	var player = get_node(str(peer_id))
+	remove_child(player)
+	player.queue_free()
```

Comment:
Careful. Better to call a function and let the player do the handling. queue_freeing directly is rarely a good idea.